### PR TITLE
add retry logic of 3 attempts for connecting to leetcode, sleep for 1…

### PR DESCRIPTION
**Pull request by kmartin62**
…0, 20 or 30 minutes respectively if it fails

Supposed to solve #32

### CODE REVIEW

**Comment by kmartin62** [comment]
@sudiptob2 Please review this.

**Comment by sudiptob2** [comment]
Thanks much :pray: will do it shortly

**Review by sudiptob2 — COMMENTED** [review]
Seems great. is it better to catch the `exception` if max retries are exceeded? I think we can do some exception handling. Like throwing `Systemexit(some message)`. What do you think?

**Comment by kmartin62** [comment]
> Seems great. is it better to catch the `exception` if max retries are exceeded? I think we can do some exception handling. Like throwing `Systemexit(some message)`. What do you think?

Great idea. Instead of `return None`, I can just `raise SystemExit("Connection to leetcode failed and max retries have been exceeded.")`. Honestly I think that the error message can get improved but I can't tell of any right now as my brain is frozen. Do you have any idea about the error message?

**Comment by sudiptob2** [comment]
Yeah, could not connect to the API or something like that would be fine I guess.

**Comment by kmartin62** [comment]
I pushed the changes. You can review them and let me know if there's still something to be changed :)

File: leeteasy/services/request_handler.py [hunk 1]
```diff
@@ -13,9 +14,19 @@ def get_challenge_info(cls) -> Union[dict, None]:
...
+                response = requests.post(url, json={'query': query})
+                return response.json().get('data').get('activeDailyCodingChallengeQuestion')
+            except Exception:
+                """
+                    On first hit sleep 10 minutes.
+                    On second hit sleep 20 minutes.
+                    On third hit sleep 30 minutes.
+                """
+                time.sleep(((i+1)*10)*60) # sleep takes seconds
+        # if it hits here, max retries were exceeded and no connection was establised
```
Comment (sudiptob2) [inline leeteasy/services/request_handler.py]: we can remove these comments, this code is self-explanatory.

**Review by sudiptob2 — COMMENTED** [review]
can you also add a unit test for this function?

**Comment by kmartin62** [comment]
> can you also add a unit test for this function?

Sadly no, I've no experience with unit tests in Python :(

**Comment by sudiptob2** [comment]
Once you push new changes, you can press the refresh button to ask for re-review. look at the screenshot. 
image

I converted the PR as a draft. Once finished and you think it's ready for merge, you can mark it as `ready for review` and I will re-review and accept.
image

Also if you can please look at some of my previous commit messages to format commit msg properly and link this issue in commit msg. Thanks much for your support :pray:

**Comment by sudiptob2** [comment]
> > can you also add a unit test for this function?
> 
> Sadly no, I've no experience with unit tests in Python :(

No problem

**Review by sudiptob2 — COMMENTED** [review]
seems good.
I have not set up **CI checks** yet but I am using flake8 for code quality.
can you run `python -m flake8` to check if there are any **flake8** errors or not in your code?

**Comment by kmartin62** [comment]
> seems good. I have not set up **CI checks** yet but I am using flake8 for code quality. can you run `python -m flake8` to check if there are any **flake8** errors or not in your code?

There was one error with having one space before an inline comment (flake8 was expecting two spaces). I fixed it and pushed, also took the commit message convention from your previous commits.

**Review by sudiptob2 — APPROVED** [review]
great work :rocket: seems perfect.

**Comment by kmartin62** [comment]
Might be two days late but I just saw that you can use `autopep8` alongside `flake8`. `autopep8` will basically format your code in pep8 style (flake8 throws an error if the style is different than pep8). You can check it here https://pypi.org/project/autopep8/

**Comment by sudiptob2** [comment]
> ake8 throws an error if the style is different th

thanks much, `autopep8` is already under my radar, But for now, I have decided not to do the formatting automatically, rather hand over this responsibility to the developer. The developer himself can use `autopep8` or any other tool to fix the formatting of his code.

**Pull request merged**